flink-core/src/main/java/org/apache/flink/util/AbstractID.java

```diff
@@ -186,7 +186,7 @@ private static long byteArrayToLong(byte[] ba, int offset) {
 		long l = 0;
 
 		for (int i = 0; i < SIZE_OF_LONG; ++i) {
-			l |= (ba[offset + SIZE_OF_LONG - 1 - i] & 0xffL) << (i << 3);
+			l |= (ba[offset + SIZE_OF_LONG - 1 - i] & 0xffL) << ((long) i << 3);
```

Comment:
I think this might be not quite right - the number of bits to shift stays below 64 and should be correct as an int.

Reply:
Agreed, I don't see any reason why this would be flagged per the Java Language Specification. The nested shift also looks to be invalid according to the Java BNF per the Java Language Specification. Will revert this change.